src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -54,25 +54,30 @@
  */
 public class BitbucketBuildStatusNotifications {
 
+    private static final String INVALID_URL_LOG_MESSAGE = "Can not determine Jenkins root URL. " +
+        "Commit status notifications are disabled until a root URL is " +
+        "configured in Jenkins global configuration.";
+
     private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener listener,
                                      @NonNull BitbucketApi bitbucket, @NonNull String hash)
             throws IOException, InterruptedException {
         JenkinsLocationConfiguration cfg = JenkinsLocationConfiguration.get();
         if (cfg == null || cfg.getUrl() == null) {
-            listener.getLogger().println(
-                    "Can not determine Jenkins root URL. Commit status notifications are disabled until a root URL is"
-                            + " configured in Jenkins global configuration.");
+            listener.getLogger().println(INVALID_URL_LOG_MESSAGE);
             return;
         }
         String url;
         try {
             url = DisplayURLProvider.get().getRunURL(build);
         } catch (IllegalStateException e) {
-            listener.getLogger().println(
-                    "Can not determine Jenkins root URL. Commit status notifications are disabled until a root URL is"
-                            + " configured in Jenkins global configuration.");
+            listener.getLogger().println(INVALID_URL_LOG_MESSAGE);
+            return;
+        }
+        if (url.startsWith("http://localhost") || url.startsWith("http://unconfigured-jenkins-location")) {
```

Comment:
You could move the if statement into the try block and just throw `IllegalStateException` 👍

Reply:
I instead extracted the logic to a private method and tried to simplify - not sure if you like it. If not I can change it again.